pkg/cmd/issue/edit/edit.go

```diff
@@ -157,59 +173,98 @@ func editRun(opts *EditOptions) error {
 	if err != nil {
 		return err
 	}
 
-	editable.Title.Default = issue.Title
-	editable.Body.Default = issue.Body
-	editable.Assignees.Default = issue.Assignees.Logins()
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	if opts.Workers == 0 {
+		opts.Workers = 10
 	}
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(opts.Workers)
+
+	issueURLs := make([]string, 0, len(issues))
+	issuesChan := make(chan string)
+	doneCh := make(chan any)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				doneCh <- nil
+				return
+			case issueURL, ok := <-issuesChan:
+				if ok {
+					issueURLs = append(issueURLs, issueURL)
+				}
+			}
+		}
+	}()
+
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating %d issues", len(issues)))
+	for _, issue := range issues {
+		issue := *issue
+		g.Go(func() error {
+			editable.Title.Default = issue.Title
+			editable.Body.Default = issue.Body
+			editable.Assignees.Default = issue.Assignees.Logins()
+			editable.Labels.Default = issue.Labels.Names()
+			editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
+			projectItems := map[string]string{}
+			for _, n := range issue.ProjectItems.Nodes {
+				projectItems[n.Project.ID] = n.ID
+			}
+			editable.Projects.ProjectItems = projectItems
+			if issue.Milestone != nil {
+				editable.Milestone.Default = issue.Milestone.Title
+			}
 
-	opts.IO.StartProgressIndicator()
-	err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+			// Allow interactive prompts for one issue; failed earlier if multiple issues specified.
+			if opts.Interactive {
+				editorCommand, err := opts.DetermineEditor()
+				if err != nil {
+					return err
+				}
+				err = opts.EditFieldsSurvey(&editable, editorCommand)
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
It feels odd allowing interactivity in the non-main Go routine. I don't know if it is an issue though. Just thought I would mention it incase that is a rule we shouldn't be breaking.

Reply:
Given the validation in `NewCmdEdit`, this can only happen when there's a single issue and stdout/stdin can be safely accessed across go routines or even threads. Given that, it seemed more disruptive to bifurcate interactive and non-interactive modes. This should be safe, but we could consider other options necessary. Resolving this just to keep tracking of pending issues, but feel free to reopen if you want to explore other options.

Reply:
I am okay with keeping it for now. I do think that bifurcating the interactive and non-interactive code paths could lead to more readable and maintainable code though, despite likely having some code duplication.